drivers/sensor/bmi088/bmi088.c

```diff
@@ -0,0 +1,313 @@
+    const struct bmi088_cfg *cfg = to_config(dev);
+    const struct spi_buf tx_buf[2] = {
+            {
+                    // Always send register address first
+                    .buf = &reg,
+                    .len = 1
+            },
+            {
+                    // Then send whatever data there is to send
+                    .buf = buf,
+                    .len = length
+            }
+    };
+    const struct spi_buf_set tx = {
+            .buffers = tx_buf,
+            // If the buffer is a null pointer, only use the first TX (register address)
+            .count = buf ? 2 : 1
+    };
+
+
+    if (write) {
+        // Write-only: No RX buffer necessary
+        return spi_write_dt(&cfg->bus, &tx);
+    } else {
+        // If we want to read, use the same buffers for that purpose.
+        // No useful data will be written to reg, as we dont receive anything while we still send the address
+        // Whatever data is in the buffer will be sent out. The sensor will ignore that data if the address was a
+        // read-request
+        const struct spi_buf_set rx = {
+                .buffers = tx_buf,
+                .count = 2
+        };
+
+        return spi_transceive_dt(&cfg->bus, &tx, &rx);
+    }
+}
+
+bool bmi088_bus_ready_spi(const struct device *dev) {
+    return spi_is_ready(&to_config(dev)->bus);
+}
+
+int bmi088_read(const struct device *dev, uint8_t reg_addr, void *buf, uint8_t len) {
+    return bmi088_transceive(dev, reg_addr | BMI088_REG_READ, false, buf, len);
+}
+
+int bmi088_byte_read(const struct device *dev, uint8_t reg_addr, uint8_t *byte) {
+    return bmi088_read(dev, reg_addr, byte, 1);
+}
+
+int bmi088_word_read(const struct device *dev, uint8_t reg_addr, uint16_t *word) {
+    int rc = bmi088_read(dev, reg_addr, word, 2);
+    if (rc != 0) {
+        return rc;
+    }
+
+    *word = sys_le16_to_cpu(*word);
+
+    return 0;
+}
+
+int bmi088_write(const struct device *dev, uint8_t reg_addr, void *buf, uint8_t len) {
+
+    return bmi088_transceive(dev, reg_addr & BMI088_REG_MASK, true, buf, len);
+}
+
+int bmi088_byte_write(const struct device *dev, uint8_t reg_addr,
+                      uint8_t byte) {
+    return bmi088_write(dev, reg_addr & BMI088_REG_MASK, &byte, 1);
+}
+
+int bmi088_word_write(const struct device *dev, uint8_t reg_addr, uint16_t word) {
+    uint8_t tx_word[2] = {
+            (uint8_t) (word & 0xff),
+            (uint8_t) (word >> 8)
+    };
+
+    return bmi088_write(dev, reg_addr & BMI088_REG_MASK, tx_word, 2);
+}
+
+int bmi088_reg_field_update(const struct device *dev, uint8_t reg_addr, uint8_t pos, uint8_t mask, uint8_t val) {
+    uint8_t old_val;
+
+    if (bmi088_byte_read(dev, reg_addr, &old_val) < 0) {
+        return -EIO;
+    }
+
+    return bmi088_byte_write(dev, reg_addr, (old_val & ~mask) | ((val << pos) & mask));
+}
+
+static void bmi088_to_fixed_point(int16_t raw_val, uint16_t scale, struct sensor_value *val) {
```

Comment:
hier bitte noch (doxygen) dokumentation was der sinn dieser funktion ist und was in/out parameter bedeuten

Reply:
und wo man gerade dabei ist: statt dem out-parameter `struct sensor_value *val` könnte man einfach einen neuen struct machen und den returnen

Reply:
Was genau wäre denn der Vorteil davon einen struct zu returnen?

Reply:
funktional ist das gleich, der vorteil liegt nur darin dass auf den ersten blick erkennbar ist, was input in diese funktion ist und was diese funktion zurückgibt. Das macht das verständnis einfacher und reduziert das fehlerpotential